models/classification_template/configs/train.yaml

```diff
@@ -0,0 +1,233 @@
+network_def:
+  _target_: DenseNet121
+  spatial_dims: 2
+  in_channels: 1
+  out_channels: '@num_classes'
+network: $@network_def.to(@device)
+
+# dataset value, this assumes a JOSN file filled with img##.nii.gz file and label
+data_json: $@bundle_root + '/data/train_samples.json'  # where training data is located and label
+data_fp: "$open(@data_json,'r', encoding='utf8')"
+data_dict: "$json.load(@data_fp)"
+partitions: '$monai.data.partition_dataset(@data_dict, (4, 1), shuffle=True, seed=0)'
+train_sub: '$@partitions[0]' # train partition
+val_sub: '$@partitions[1]' # validation partition
+
+# these transforms are used for training and validation transform sequences
+base_transforms:
+- _target_: LoadImaged
+  keys: '@image'
+- _target_: EnsureChannelFirstd
+  keys: '@image'
+
+# these are the random and regularising transforms used only for training
+train_transforms:
+- _target_: RandAxisFlipd
+  keys: '@image'
+  prob: '@rand_prob'
+- _target_: RandRotate90d
+  keys: '@image'
+  prob: '@rand_prob'
+- _target_: RandGaussianNoised
+  keys: '@image'
+  prob: '@rand_prob'
+  std: 0.05
+- _target_: ScaleIntensityd
+  keys: '@image'
+
+# these are used for validation data so no randomness
+val_transforms:
+- _target_: ScaleIntensityd
+  keys: '@image'
+
+# define the Compose objects for training and validation
+preprocessing:
+  _target_: Compose
+  transforms: $@base_transforms + @train_transforms
+
+val_preprocessing:
+  _target_: Compose
+  transforms: $@base_transforms + @val_transforms
+
+# define the datasets for training and validation
+train_dataset:
+  _target_: Dataset
+  data: '@train_sub'
+  transform: '@preprocessing'
+
+val_dataset:
+  _target_: Dataset
+  data: '@val_sub'
+  transform: '@val_preprocessing'
+
+# define the dataloaders for training and validation
+train_dataloader:
+  _target_: ThreadDataLoader  # generate data ansynchronously from training
+  dataset: '@train_dataset'
+  batch_size: '@batch_size'
+  repeats: '@num_substeps'
+  num_workers: '@num_workers'
+
+val_dataloader:
+  _target_: DataLoader  # faster transforms probably won't benefit from threading
+  dataset: '@val_dataset'
+  batch_size: '@batch_size'
+  num_workers: '@num_workers'
+
+# Simple CrossEntropy loss configured for multi-class classification
+lossfn:
+  _target_: torch.nn.CrossEntropyLoss
+  reduction: sum
+
+# hyperparameters could be added for other arguments of this class
+optimizer:
+  _target_: torch.optim.Adam
+  params: $@network.parameters()
+  lr: '@learning_rate'
+
+# should be replaced with other inferer types if training process is different for your network
+inferer:
+  _target_: SimpleInferer
+
+# transform to apply to data from network to be suitable for validation
+postprocessing:
+  _target_: Compose
+  transforms:
+  - _target_: Activationsd
+    keys: '@pred'
+    softmax: true
+  - _target_: AsDiscreted
+    keys: ['@pred', '@label']
+    argmax: [true, false]
```

Comment:
Same thing I suggested above. If this version of `postprocessing` is identical to that in `evaluate.yaml` then it doesn't have to be repeated there I don't think.